widgetssdk/src/main/java/com/glia/widgets/callvisualizer/controller/CallVisualizerController.kt

```diff
@@ -107,7 +108,7 @@ internal class CallVisualizerController(
 
     private fun onIncomingEngagementRequestTimeout() {
         dialogController.dismissCVEngagementConfirmationDialog()
-        closeHolderActivity()
+        _state.onNext(CallVisualizerContract.State.ShowTimeoutSnackBar)
```

Comment:
I would recommend keeping the `closeHolderActivity()` call here before pushing a new state instead of moving it to the ActivityWatcher.
Each state should be responsible for its action like closing holder activity, showing snackbar, etc.

Reply:
Unfortunately, it doesn't work. It isn't a stack of events. Only the last event will apply if the application was in the background during the timeout.

Reply:
I can only suggest renaming the state to `CloseHolderActivityAndShowTimeoutSnackBar`.